### ISSUE TITLE
Review and improve the ML quiz questions

### ISSUE THREAD

**Issue by dalelane** [IBM/taxinomitis#22]
The Models UI page displays a machine learning concepts multi-choice quiz for students to do while they're waiting for a model to train. This is particularly useful for images projects which can take 4-5 minutes to finish training a model sometimes. 

The quiz questions were written in a hurry for the initial site release, with the plan that I'd come back and write some proper ones.

Except I mostly never did.

I should double-check if they're as bad as I suspect.

Adding new ones would be good, too - as I've seen students get through all of them and still have a minute to wait.

**Comment by dalelane** [comment]
ignore the commit above.... in moving the repo from Github Enterprise to public Github, the issue numbers don't match up, so it refers to a closed issue from the old repo :(

**Comment by dalelane** [comment]
The quiz questions live here:
https://github.com/IBM/taxinomitis/blob/390954da60e4c75d2a13db911ddd8300f33ab826/public/components/quiz/quiz.service.js#L13-L136

**Comment by ravenscroftj** [comment]
Let me prefix everything here by saying that I recently got off a 14 hour flight so maybe my analysis isn't as spot on as it should be...

Generally I like the questions that you already have in the system. They’re interesting and practical and I think they’re relatively understandable. I don’t think they’re as bad as perhaps you suspect.

You cover:
* Training data variance
* Balanced and unbalanced datasets
* Limits of supervised learning
* Rules of thumb and practicalities for data collection
* Good and bad use cases for machine learning

I think there’s definitely stuff missing that would be interesting. I've come up with a few suggestions:

### More stuff on training data and human bias
There are a couple of examples of this already but I think it could be extended and we can even introduce the concept of inter-annotator-agreement in simplistic terms. 

-----------

**How could you make sure that your collection of animal pictures is good quality for training a machine and doesn’t contain human mistakes?**

a) Manually inspect all of the pictures of animals myself and make sure I agree with all of the labels
b) Have more than one person label each image and keep whichever label people generally agree upon
c) Trick question! Humans are always right - machines make mistakes!
d) Only let your friends and people you trust label the images

**Notes:**
Even humans make mistakes and you’ll never have time to double check all the data! If the task is tricky it can be a good idea to get more than one human expert to label each image and keep the label that the most humans agree is the right one.

------------------

**A machine trained to identify pictures of ‘food' or 'not food’ gives inconsistent results for sandwiches, often putting them in the “not food” class. Which of the following could explain this behaviour?**

**a) The human who trained the system doesn’t like sandwiches and didn’t put any in the training dataset.**
b) The machine has to be in the right mood for a sandwich so it changes its mind a lot.
c) Sandwiches are not food. The system is right to be unsure.

**Notes**
Machine learning models can only learn from the examples we show them. It is very easy for humans to accidentally introduce their likes and dislikes into models so we should be very careful about what training data we use. 

----------------------

### Other Evaluation Metrics
It may not be necessary to introduce kids to Precision, Recall and F1-Score but it might be nice to get them to question over-relliance on accuracy.

**A rock-paper-scissors model has 100 pictures of rock, 25 pictures of paper and 25 pictures of scissors. It is 90% accurate after it is trained. Which statement is true?**

a) The classifier is very good at predicting all classes
b) The classifier is very good at predicting paper
c) The classifier is very good at predicting scissors
**d) It’s difficult to know how good the classifier is from its overall accuracy because the training set was imbalanced**

**Notes**
Sometimes accuracy is not very helpful for measuring how good a model is. Especially when the model predicts many classes or there’s an imbalanced training set. Try reading about Precision and Recall as a more helpful way to measure performance in these cirumstances.

---------------------

### Transparency and Black Box Models

It might be good to teach the kids about how most of these models are “black boxes” and its difficult to reason about the decisions they make. Perhaps introducing them to decision trees as a “transparent” model.

**Which of the following is a weakness of advanced ‘neural network’ machine learning?**

a) Neural networks take a long time to train and its boring
b) Neural networks only work on images
**c) It’s very difficult to understand why a neural made a decision after it has been trained**
d) You have to use a graphics card (GPU) to train a neural network.

**Notes**
While it’s true that neural networks take a while to train and that GPUs can make them faster, not knowing why a neural network made a decision is a big problem that makes it hard to trust their results in important scenarios (like diagnosing types of diseases or letting people borrow money from banks).

-------------------------

**Which of the following machine learning techniques can tell you why it made a choice?**

a) Neural Networks (Uses neurons and synapses to make a decision)
b) Naive Bayes (Uses probability to work out the most likely class)
**c) Decision Trees (branches represent features that help it classify data)**
d) Logistic Regression (Plots data on a graph and draws a line to split it into classes)

**Notes**
Decision trees learn clear, explainable rules from data and always follow them. The decisions that they make are very easy to explain but they are not always as accurate as other methods.


-------------

Perhaps some more general “spot the odd one out” questions to dispel more myths about machine learning. I'll have a bit of a think.

**Comment by dalelane** [comment]
@ravenscroftj that's really useful, thanks!
